lang/types.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Asserts that the line of code is never executed.
+ *
+ * This is useful when you want to force a compiler error if a switch statement is not exhaustive.
+ *
+ * @example
+ * switch (architecture) {
+ *   case 'arm64':
+ *     return false;
+ *   case 'x86_64':
+ *     return true;
+ *   default:
+ *     checkExhaustive(architecture);
+ * }
+ */
+export function checkExhaustive<T = unknown>(value: never): T {
```

Comment:
I think the types here might be backwards? Should this be `checkExhaustive<T = unknown>(value: T): never`?

(Related, I'm curious about why we need the generic type. Could the argument value just be an `any`?